jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtil.java

```diff
@@ -177,4 +177,19 @@ public static String getProperty(String key, JavaProject project) {
     }
     return value;
   }
+
+  public static File resolveArtifact(JavaProject project, String groupId, String artifactId, String version, String type) {
+    File artifact = project.getDependencies().stream()
+        .filter(d -> d.getGroupId().equals(groupId)
+            && d.getArtifactId().equals(artifactId)
+            && d.getVersion().equals(version)
+            && d.getType().equals(type))
+        .findFirst()
+        .map(Dependency::getFile)
+        .orElse(null);
+    if (artifact == null) {
+      throw new IllegalStateException("Cannot find artifact " + String.format("%s-%s.%s", artifactId, version, type) + " within the resolved resources");
+    }
+    return artifact;
+  }
```

Comment:
We already have a getDependency and getPlugin.
Do we really need another method for this?
If yes, is the exception thrown here appropriate or something we want?

Reply:
It's used by two classes: SpringBootGenerator and SpringBootWatcher, I can move this code to these calling places if you want.